modules/azure_aks/variables.tf

```diff
@@ -84,21 +84,41 @@ variable "aks_dns_service_ip" {
   description = "IP address within the Kubernetes service address range that will be used by cluster service discovery (kube-dns). Changing this forces a new resource to be created."
   type        = string
   default     = "10.0.0.10"
+  validation {
+    condition     = var.aks_dns_service_ip != null ? can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",var.aks_dns_service_ip)) : false
+    error_message = "ERROR: aks_dns_service_ip - value must not be null and must be a valid IP address."
+  }
+
 }
 
 variable "aks_docker_bridge_cidr" {
   description = "IP address (in CIDR notation) used as the Docker bridge IP address on nodes. Changing this forces a new resource to be created."
   default     = "172.17.0.1/16"
+  validation {
+    condition     = var.aks_docker_bridge_cidr != null ? can(cidrnetmask(var.aks_docker_bridge_cidr)) : false
+    error_message = "ERROR: aks_docker_bridge_cidr - value must not be null and must be valid CIDR."
+  }
+
 }
 
 variable "aks_pod_cidr" {
   description = "The CIDR to use for pod IP addresses. This field can only be set when network_plugin is set to kubenet. Changing this forces a new resource to be created."
   default     = "10.244.0.0/16"
+  validation {
+    condition     = var.aks_pod_cidr != "" ? can(cidrnetmask(var.aks_pod_cidr)) : true
```

Comment:
For aks_pod_cirdr, I believe it is a valid configuration for it to be empty when not using the kubenet plugin. So I don't think this line should change, as the possible outcomes from this method are the following:

1) The customer's non empty valid input. That is used.
2) The customer's non empty invalid input. The can(cidrnetmask) catches this and throws an error.
3) The customer's empty input. This is used since it is valid.
4) The default value, which occurs when the customer passes null.